Dockerfile

```diff
@@ -0,0 +1,16 @@
+FROM python:3.4-slim
+WORKDIR /app
+COPY ./images /app/images
+COPY ./libgrabsite /app/libgrabsite
+COPY ./grab-site ./gs-dump-urls ./gs-server ./setup.py /app/
+RUN apt-get update
+RUN apt-get install -y build-essential
+RUN pip3 install ./
+RUN apt-get purge -y build-essential
+RUN apt-get autoremove -y
+RUN apt-get clean
```

Comment:
Another possibility is to use `FROM python:3.4` instead of `FROM python:3.4-slim`; the non-slim variant is based off of `buildpack-deps` which has a lot of compilers / tools / libraries installed. The resulting total image size would be bigger, but the advantage is that the `buildpack-deps` portion would be shared with every other image based off of that, so in the usual case where you have several images, the total space usage would be lower.